internal/operands/metrics/resources.go

```diff
@@ -139,15 +149,123 @@ func getRecordRules() []promv1.Rule {
 	return recordRules
 }
 
+func newMonitoringRbacRole(namespace string) *rbac.Role {
+	return &rbac.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusRole,
+			Namespace: namespace,
+		},
+		Rules: []rbac.PolicyRule{{
+			APIGroups: []string{""},
+			Resources: []string{"services", "endpoints", "pods"},
+			Verbs:     []string{"get", "list", "watch"},
+		}},
+	}
+}
+
+func newMonitoringRbacRoleBinding(namespace string) *rbac.RoleBinding {
+	return &rbac.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusRole,
+			Namespace: namespace,
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      prometheusServiceAccountName,
+				Namespace: MonitorNamespace,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			Kind:     "Role",
+			Name:     prometheusRole,
+			APIGroup: rbac.GroupName,
+		},
+	}
+}
+
+func newServiceMonitorCR(namespace string) *promv1.ServiceMonitor {
+	return &promv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: monitoring.GroupName,
+			Kind:       "ServiceMonitor",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      PrometheusRuleName,
+			Labels: map[string]string{
+				"openshift.io/cluster-monitoring": "",
+				prometheusLabelKey:                prometheusLabelValue,
+				"k8s-app":                         "kubevirt",
+			},
+		},
+		Spec: promv1.ServiceMonitorSpec{
+			NamespaceSelector: v1.NamespaceSelector{
+				MatchNames: []string{namespace, "kubevirt"},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					prometheusLabelKey: prometheusLabelValue,
+				},
+			},
+			Endpoints: []promv1.Endpoint{
+				{
+					Port:   "metrics",
```

Comment:
Can you extract this port name into a constant? It is used in multiple files.

Reply:
done